<a href="https://colab.research.google.com/github/Aashrit77/Assignments/blob/main/Hypoassi4.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [18]:
import pandas as pd
import numpy as np
from scipy import stats
from scipy.stats import chi2_contingency

In [19]:
data = pd.read_csv("/content/Costomer+OrderForm.csv")
data.head()

Phillippines   Indonesia       Malta       India
0   Error Free  Error Free   Defective  Error Free
1   Error Free  Error Free  Error Free   Defective
2   Error Free   Defective   Defective  Error Free
3   Error Free  Error Free  Error Free  Error Free
4   Error Free  Error Free   Defective  Error Free

In [21]:
data.Phillippines.value_counts()

Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [22]:
data.Indonesia.value_counts()

Error Free    267
Defective      33
Name: Indonesia, dtype: int64

In [23]:
data.Malta.value_counts()

Error Free    269
Defective      31
Name: Malta, dtype: int64

In [25]:
data.India.value_counts()

Error Free    280
Defective      20
Name: India, dtype: int64

In [27]:
array = np.array([[271,267,269,280],[29,33,31,20]])
array

array([[271, 267, 269, 280],
       [ 29,  33,  31,  20]])

Assume Null Hypothesis H0 = defective% does not change by center
Alternative Hypothesis Ha = defective% varies by center

In [28]:
chi2_contingency(array)

(3.858960685820355,
 0.2771020991233135,
 3,
 array([[271.75, 271.75, 271.75, 271.75],
        [ 28.25,  28.25,  28.25,  28.25]]))

As P-Value(0.277)>0.05 accept H0 (Defective% does not varies by center